optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/SolverProblemBenchmarkResult.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.benchmark.impl.result;
+
+import java.io.File;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.config.solver.SolverConfig;
+
+public interface SolverProblemBenchmarkResult {
```

Comment:
This interface was created due to Single Statistics, at first. Then it became apparent, that we need something representing a "SolverProblemBenchmark" (anything that can be described as a cross section of a Problem and Solver Benchmark - Singles and Subsingles): when calculating the standard deviation of scores, in the report when adding badges.

`SolverProblemBenchmarkResult` was the best thing I came up with, as it fits into the model quite nicely. I'm not sure if we want to have more of them, as there's no necessary need so far. As for naming, I agree that is debatable, although I'm not sure that SubSingleBenchmark implementing SingleBenchmark and DefaultSingleBenchmark implementing SingleBenchmark is the way to go, but it's not necessarily bad.